spec/factories/course_assessment_categories.rb

```diff
@@ -2,7 +2,7 @@
   sequence(:course_assessment_category_weight) { |n| n }
   sequence(:course_assessment_category_title) { |n| "Category #{n}" }
   factory :course_assessment_category, class: Course::Assessment::Category do
-    course
+    course { create(:course) }
```

Comment:
Because this method was called in `before_validation`, it requires root_folder but that time course haven't been created. https://github.com/coursemology-collab/coursemology2/blob/wangqiang/category-callbacks/app/models/course/assessment/category.rb#L42